topostats/plotting.py

```diff
@@ -177,18 +182,18 @@ def sns_plot(self) -> None:
         fig, ax = self._setup_figure()
         if self.hist and not self.kde:
             outfile = self._outfile("hist")
-            sns.histplot(data=self.melted_data, x="value", bins=self.bins, stat=self.stat, hue=self.image_id)
+            sns.histplot(data=self.melted_data, x="value", bins=self.bins, stat=self.stat, hue="basename")
```

Comment:
Hard coding this makes it tricky if people then want to colour/group by different variables.

I think the class `__init__` should take a value for `hue` and in some other plots `x` which become properties. Defaults should be `basename` but its then allows users to specify different groupings should they wish to.

Reply:
I think I've addressed this as you suggested, but please do check if I've misinterpreted or missed something